src/main/java/io/r2dbc/pool/ConnectionPool.java

```diff
@@ -163,27 +164,50 @@ public void close() {
         dispose();
     }
 
+    /**
+     * Dispose this {@link ConnectionPool} in non-blocking flow.
+     * <p>
+     * When multiple errors occurred during dispose flow, they are added as
+     * suppressed errors onto the first error.
+     *
+     * @return a Mono triggering the shutdown of the pool once subscribed.
+     */
+    public Mono<Void> disposeLater() {
```

Comment:
Related to r2dbc/r2dbc-spi#91: Let's align these methods with the proposed `Closeable`. I'm not sure whether `Closeable.close()` will remain as proposed or whether we will see changes to the signature. 

The pool implements `Disposable` because there's no SPI-based convention on how to release associated resources. Final method signatures of this PR depend pretty much on the outcome of r2dbc/r2dbc-spi#91.

Reply:
Ok, so this method name will be `close` once r2dbc/r2dbc-spi#91 gets finalized, or make `disposeLater()` `private` and the new method just delegate to this.
And if the method name becomes `close`, then we'll remove(?) `java.io.Closeable#close` since it conflicts the method name.

Reply:
> Ok, so this method name will be close once r2dbc/r2dbc-spi#91 

Exactly.

Please note that SPI specifies its own `Closeable` that is not `java.io.Closeable`. SPI's `close()` returns `Publisher<Void>` to allow resource disposal in a non-blocking way. Using `java.io.Closeable#close` would either require blocking or would not allow for synchronization and these are the two reasons why SPI should come up with its own `Closeable` interface.